std/conv.d

```diff
@@ -3880,23 +3860,28 @@ unittest
     static assert(!__traits(compiles, octal!"spam"));
     static assert(!__traits(compiles, octal!"77%"));
 
+    static assert(is(typeof(octal!"17777777777") == int));
+    static assert(octal!"17777777777" == int.max);
+
+    static assert(is(typeof(octal!"20000000000U") == ulong)); // Shouldn't this be uint?
+    static assert(octal!"20000000000" == uint(int.max) + 1);
+
+    static assert(is(typeof(octal!"777777777777777777777") == long));
+    static assert(octal!"777777777777777777777" == long.max);
+
+    static assert(is(typeof(octal!"1000000000000000000000U") == ulong));
+    static assert(octal!"1000000000000000000000" == ulong(long.max) + 1);
+
     int a;
     long b;
 
     // biggest value that should fit in an it
-    static assert(__traits(compiles,  a = octal!"17777777777"));
+    static assert(__traits(compiles, a = octal!"17777777777"));
     // should not fit in the int
     static assert(!__traits(compiles, a = octal!"20000000000"));
     // ... but should fit in a long
     static assert(__traits(compiles, b = octal!"20000000000"));
 
-    static assert(!__traits(compiles, a = octal!"1L"));
-
-    // this should pass, but it doesn't, since the int converter
-    // doesn't pass along its suffix to helper templates
```

Comment:
This was never true. I'm pretty sure the reason it compiles is because the compiler knows that octal!1L fits in an `int`.